botorch/optim/optimize_homotopy.py

```diff
@@ -199,14 +209,16 @@ def optimize_acqf_homotopy(
             ).unsqueeze(1)
 
         # Optimize one more time with the final options
-        candidates, acq_values = optimize_acqf(
-            acq_function=acq_function,
-            bounds=bounds,
-            q=1,
-            options=final_options,
-            batch_initial_conditions=candidates,
-            **shared_optimize_acqf_kwargs,
-        )
+        if fixed_features_list:
```

Comment:
I don't think you want to make things conditional on this - we did this even before supporting `fixed_features_list`

Reply:
Good catch. Thanks! I overlooked the if when I was changing it to `optimization_fn`.